services/openvpn/service/manager.go

```diff
@@ -48,6 +49,7 @@ type SessionConfigNegotiatorFactory func(secPrimitives *tls.Primitives, outbound
 // Manager represents entrypoint for Openvpn service with top level components
 type Manager struct {
 	natService nat.NATService
+	mapperQuit chan struct{}
```

Comment:
I'm not sure if the transport service should be aware of the port mapper. Maybe it should be handled on some upper layer?

Reply:
Actualy instead of passing something like port mapping, what about exposing interface where service can request port pair allocation - one for passing to consumer (in session config) another for binding transport itself. It can then close or release that port pair, notifying port mapper that's it's no longer needed. Bonus points - in case of binding on public ip you can pass noop implemention - which simply always returns preconfigured ports.

Reply:
I don't see a point to abuse DI and lift it somewhere else since it is not stand alone service as such, but maintains mapping just for concrete service and such maintenance goes down together with a service. I really don't think that there is a point making it self sustained service, lets leave it as simple as possible.

Reply:
The only point here - hide implementation details. If services needs publically accesible port - just give it to him, nothing more nothing less

Reply:
As per @tadovas suggestion. I do agree that we need dynamic port handling. That is service should request some randomly generated port from local resource pool. But this relates to many parts of the code and imho should not be addressed in this PR. Created #689 ticket for this.